### PR TITLE
Gasg pipeline

### PR DESCRIPTION
**Pull request by jamalxo**
This branch added the support for geometry aware sparse grids to the pipeline. Changes have only been made in the /datadriven module.

### CODE REVIEW

File: datadriven/src/sgpp/datadriven/configuration/GeometryConfiguration.hpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/*
+ * GeometryConfiguration.hpp
+ *
+ *  Created on: Jan 15, 2019
+ *      Author: jamal
+ */
+
+#ifndef DATADRIVEN_SRC_SGPP_DATADRIVEN_CONFIGURATION_GEOMETRYCONFIGURATION_HPP_
```
Comment (krsgpp) [inline datadriven/src/sgpp/datadriven/configuration/GeometryConfiguration.hpp]: Please use #pragma once

File: datadriven/src/sgpp/datadriven/datamining/base/SparseGridMinerSplitting.cpp [hunk 2]
```diff
@@ -50,6 +50,8 @@ double SparseGridMinerSplitting::learn(bool verbose) {
       // Train model on new batch
       fitter->update(*dataset);
 
+      std::cout << "Done training!" << std::endl;
```
Comment (krsgpp) [inline datadriven/src/sgpp/datadriven/datamining/base/SparseGridMinerSplitting.cpp]: This output should be removed again.

**Comment by krsgpp** [comment]
Boost tests are still missing.

File: datadriven/src/sgpp/datadriven/algorithm/GridFactory.cpp [hunk 3]
```diff
@@ -26,32 +28,93 @@ using sgpp::base::algorithm_exception;
...
+  } else {
+    std::cout << "Creating geometry aware sparse grid..." << std::endl;
+    tmpGrid->getGenerator().regularInter(gridConfig.level_, interactions, 0.0);
+    std::cout << "Interactions set!" << std::endl;
+  }
+  return tmpGrid;
+}
+
+std::vector<std::vector<size_t>> sgpp::datadriven::GridFactory::getInteractions(
+    std::string& stencil, std::vector<int64_t>& dim) const {
```
Comment (krsgpp) [inline datadriven/src/sgpp/datadriven/algorithm/GridFactory.cpp]: stencil should be an enum.

Comment (jamalxo) [reply datadriven/src/sgpp/datadriven/algorithm/GridFactory.cpp]: yes it can be, but it would lead in to changing every method again

**Comment by jamalxo** [comment]
> Boost tests are still missing.

how do I create a small test dataset to test the gasg pipeline integration?

File: datadriven/src/sgpp/datadriven/algorithm/GridFactory.cpp (hunk 3 above)
Comment (krsgpp) [reply datadriven/src/sgpp/datadriven/algorithm/GridFactory.cpp]: Do it please.

**Comment by krsgpp** [comment]
> > Boost tests are still missing.
> 
> how do I create a small test dataset to test the gasg pipeline integration?

Look at datadriven/datasets, there are some very small datasets in there already.

**Comment by jamalxo** [comment]
> > > Boost tests are still missing.
> > 
> > 
> > how do I create a small test dataset to test the gasg pipeline integration?
> 
> Look at datadriven/datasets, there are some very small datasets in there already.

Using the dataread dataset I get the following error

###############Starting training epoch #0
###############Itertation #0
Batch size: 4
Registered new class label -5 with index 0
Dataset dimension 5
Creating geometry aware sparse grid...
**invalid fastbin entry (free)**   <-----
**Aborted (core dumped)**      <-----


Do you know why this is the case?

File: datadriven/examplesPipeline/geometryAwareSparseGrid.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
+{
+  	"dataSource": {
+    		"filePath": "/home/jamal/Documents/SGpp/datasets/mnist/mnist_[2]_28x28.train.arff",
```
Comment (krsgpp) [inline datadriven/examplesPipeline/geometryAwareSparseGrid.json]: Why did you change that to an absolute path?

Comment (jamalxo) [reply datadriven/examplesPipeline/geometryAwareSparseGrid.json]: my mistake I will fix it asap

**Comment by jamalxo** [comment]
I can add the boost tests this weekend, however I still need to find out why it doesnt create a grid for small datasets

File: datadriven/tests/DataMiningConfigParserTest.cpp [hunk 5]
```diff
@@ -18,8 +18,10 @@
 #include <sgpp/datadriven/datamining/modules/dataSource/DataSourceConfig.hpp>
 #include <sgpp/datadriven/datamining/modules/dataSource/DataTransformationConfig.hpp>
 #include <sgpp/datadriven/datamining/modules/scoring/ScorerConfig.hpp>
+#include <sgpp/datadriven/configuration/GeometryConfiguration.hpp>
```
Comment (krsgpp) [inline datadriven/tests/DataMiningConfigParserTest.cpp]: Please sort includes alphabetically

**Comment by krsgpp** [comment]
Besides the open discussions, please also take a look at the errors and warnings from the jenkins jobs: https://simsgs.informatik.uni-stuttgart.de/jenkins/job/SGpp-meta-testing/ and commit the fixes.

**Comment by jamalxo** [comment]
> Besides the open discussions, please also take a look at the errors and warnings from the jenkins jobs: https://simsgs.informatik.uni-stuttgart.de/jenkins/job/SGpp-meta-testing/ and commit the fixes.

for the latest build #30 i didnt see any warning or errors

**Comment by jamalxo** [comment]
> > Besides the open discussions, please also take a look at the errors and warnings from the jenkins jobs: https://simsgs.informatik.uni-stuttgart.de/jenkins/job/SGpp-meta-testing/ and commit the fixes.
> 
> for the latest build #30 i didnt see any warning or errors



> > Besides the open discussions, please also take a look at the errors and warnings from the jenkins jobs: https://simsgs.informatik.uni-stuttgart.de/jenkins/job/SGpp-meta-testing/ and commit the fixes.
> 
> for the latest build #30 i didnt see any warning or errors

Nvm. I found this
-----
The following parameters of sgpp::datadriven::ModelFittingBase::buildGrid(const RegularGridConfiguration &gridConfig, const GeometryConfiguration &geometryConfig) const are not documented: parameter 'geometryConfig'
-----
How do I document that parameter or what is the point of this warning?
